tests/test_api_requests.py

```diff
@@ -310,3 +506,742 @@ async def parse(self, response):
+            {},
+            {
+                "browserHtml": True,
+                "httpResponseHeaders": True,
+            },
+            [],
+        ),
+        (
+            {
+                "headers": {
+                    "Accept": "application/json",
+                    "Accept-Language": "uk",
+                },
+                "meta": {"zyte_api": {"browserHtml": True}},
+            },
+            {},
+            {
+                "browserHtml": True,
+                "httpResponseHeaders": True,
+            },
+            ["cannot be mapped"],
+        ),
+        # The Cookie header is dropped with a warning.
+        (
+            {
+                "headers": {
+                    "Cookie": "a=b",
+                },
+            },
+            {},
+            {
+                "httpResponseBody": True,
+                "httpResponseHeaders": True,
+            },
+            ["cannot be mapped"],
+        ),
+        (
+            {
+                "headers": {
+                    "Cookie": "a=b",
+                },
+                "meta": {"zyte_api": {"browserHtml": True}},
+            },
+            {},
+            {
+                "browserHtml": True,
+                "httpResponseHeaders": True,
+            },
+            ["cannot be mapped"],
+        ),
+        # The User-Agent header, which Scrapy sets by default, is dropped
+        # silently if it matches the default value of the USER_AGENT setting,
+        # or with a warning otherwise.
+        (
+            {
+                "headers": {"User-Agent": DEFAULT_USER_AGENT},
+            },
+            {},
+            {
+                "httpResponseBody": True,
+                "httpResponseHeaders": True,
+            },
+            [],
+        ),
+        (
+            {
+                "headers": {"User-Agent": ""},
+            },
+            {},
+            {
+                "httpResponseBody": True,
+                "httpResponseHeaders": True,
+            },
+            ["cannot be mapped"],
+        ),
+        (
+            {
+                "headers": {"User-Agent": DEFAULT_USER_AGENT},
+                "meta": {"zyte_api": {"browserHtml": True}},
+            },
+            {},
+            {
+                "browserHtml": True,
+                "httpResponseHeaders": True,
+            },
+            [],
+        ),
+        (
+            {
+                "headers": {"User-Agent": ""},
+                "meta": {"zyte_api": {"browserHtml": True}},
+            },
+            {},
+            {
+                "browserHtml": True,
+                "httpResponseHeaders": True,
+            },
+            ["cannot be mapped"],
+        ),
+        # You may update the ZYTE_API_UNSUPPORTED_HEADERS setting to remove
+        # headers that the customHttpRequestHeaders parameter starts supporting
+        # in the future.
+        (
+            {
+                "headers": {"User-Agent": ""},
```

Comment:
Should we add the `Cookie` header here to test out `ZYTE_API_UNSUPPORTED_HEADERS`?